### DISCUSSION TITLE
Taro 支持 react-router & vue-router ，它来了

### DISCUSSION THREAD

**Discussion by AdvancedCat** [NervJS/taro#11800] [RFC]
## 背景

Taro 收集了不少来自开发者的呼声，希望能够支持 `history & location` ，如此就可以在 Taro 中使用前端成熟的路由方案，如 `react-router & vue-router`，从此在 Taro 中开发单页面应用成为可能。

详细内容可查看 [rfc-history](https://github.com/NervJS/taro-rfcs/blob/feat/history/rfcs/0009-router-support.md)

## 分支

feat/history

## 如何尝鲜

首先，你需要 `clone taro` 至本地：
```bash
# In taro repo
$ cd taro
$ git checkout -b feat/history origin/feat/history
$ yarn
$ yarn build
$ lerna link

# In local project
$ cd MyProject
$ yarn link @tarojs/cli
$ npm run dev:weapp
```
> 在功能稳定后，会发布 alpha 版本

## 反馈

因 Taro 维护了最小化 history 实现，难免会有一些未发现的问题，如有任何建议或疑问，可以在此 discussion 下反馈。
欢迎感兴趣者参与共建，也请联系我。

**Comment by Airkro** [comment]
`react-router / vue-router` 都有提供路由守卫的功能，利用它可以减少H5和小程序的差异

一个平替方法，欢迎讨论：https://github.com/NervJS/taro-rfcs/issues/22
